src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -793,6 +794,16 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP138",
                 "Loops are not currently supported.");
+       
+            public ErrorDiagnostic InvalidResourceTypePart(string typePart) => new(
+                TextSpan,
+                "BCP139",
+                $"The resource type part \"{typePart}\" is invalid. Nested resources must specify a single type, and optionally can specify a version using the format \"<type>@<apiVersion>\".");
+
+            public ErrorDiagnostic InvalidAncestorResourceType(string resourceName) => new(
+                TextSpan,
+                "BCP140",
+                $"The resource type cannot be determined due an error in containing resource \"{resourceName}\".");

```

Comment:
Is it possible to be more specific here about what the error is (invalid type?), or is this just handling the general parent symbol == error symbol scenario?

Reply:
I'm not sure if we could be more specific, because this is handling parent symbol == error symbol or parent type == error type.

Since we validate the parent/child relationship based on the naming structure, this doesn't happen when the parent has a valid but unknown type.